domain-models-runtime/src/test/java/org/folio/rest/persist/GeneralFunctionsIT.java

```diff
@@ -21,22 +21,24 @@ public class GeneralFunctionsIT extends PostgresClientITBase {
 
   private Future<Row> upsert(TestContext context, UUID id, JsonObject json) {
     PostgresClient postgresClient = PostgresClient.getInstance(vertx, tenant);
-    String sql = "SELECT upsert('t', $1, $2)";
+    String sql = "SELECT upsert('t', $1, $2);";
     return postgresClient.selectSingle(sql, Tuple.of(id, json))
-    .onComplete(context.asyncAssertSuccess(upsert -> {
-      assertThat(sql, upsert.getUUID(0), is(id));
-      postgresClient.getById("t", id.toString(), context.asyncAssertSuccess(get -> {
-        assertThat(get, is(json));
-      }));
-    }));
+        .onComplete(context.asyncAssertSuccess(upsert -> {
+          assertThat(sql, upsert.getUUID(0), is(id));
+          postgresClient.getSQLReadConnection(conn -> {
+            postgresClient.getById(conn, "t", id.toString(), context.asyncAssertSuccess(get -> {
+              assertThat(get, is(json));
+            }));
+          });
+        }));
```

Comment:
The strangeness might be due to [what I observed in my ](https://github.com/folio-org/raml-module-builder/pull/1123/files#diff-7dce7229cecb42a85babe8bc265ecd46f1f32cfa02089ff29a7e3cdfcba6ec43R112)`testReadWrite` test. Even with sync configured, the first write has to have that `synchronous_commit` flag set in a transaction in that test. We're starting up `PostgresTesterContainer` quite a few times where this could catch us. But once the first data replicates, then subsequent operations are fully synchronous. Probably because the  WAL has to be "woken up" or something in the first transaction. In any case the solution is probably to force the test to use the read/write conn either way.

Reply:
The test doesn't pass without my change in sync or async mode. I don't fully understand why. The only difference between the code I have, and the original code that I can see is that the one that works news up the `Conn` object if I'm remembering right.